lib/core/src/Cardano/Wallet/Api/Server.hs

```diff
@@ -698,9 +699,25 @@ mkShelleyWallet ctx wid cp meta pending progress = do
         -- never fails - returns zero if balance not found
         liftIO $ fmap fromIntegral <$> W.fetchRewardBalance @_ @s @k wrk wid
 
+    let ti = timeInterpreter $ ctx ^. networkLayer @t
 
+    -- In the Shelley era of Byron;Shelley;Allegra toApiWalletDelegation using
+    -- an unextended @ti@ will simply fail because of uncertainty about the next
+    -- fork.
+    --
+    -- @unsafeExtendSafeZone@ performs the calculation as if no fork will occur.
+    -- This should be fine because:
+    -- 1. We expect the next few eras to have the same epoch length as Shelley
+    -- 2. It shouldn't be the end of the world to return the wrong time.
+    --
+    -- But ultimately, we might want to make the uncertainty transparent to API
+    -- users.
```

Comment:
Perhaps link this comment to a JIRA technical-debt ticket or this will never happen?

Reply:
Done -> ADP-575